handlers.py

```diff
@@ -59,69 +64,71 @@ async def registration(message: types.Message):
 
 
 @dp.message_handler(lambda message: message.text == "Register")
-async def registration(message: types.Message):
-    await Form.name.set()
-    await message.answer('Was ist deine Vorname?')
-
-
-@dp.message_handler(state=Form.name)
-async def process_name(message: types.Message, state: FSMContext):
-    async with state.proxy() as data:
-        data['name'] = message.text
-    await message.answer('Was ist deine Familienname?')
-    await Form.next()
-
+async def register(message: types.Message):
+    keyboard = types.ReplyKeyboardMarkup(resize_keyboard=True, one_time_keyboard=True)
+    button = types.KeyboardButton("Share a contact", request_contact=True)
+    keyboard.add(button)
+    await bot.send_message(chat_id=message.from_user.id, text="In order to use a bot, please share your contact!", reply_markup=keyboard)
 
-@dp.message_handler(state=Form.surname)
-async def process_surname(message: types.Message, state: FSMContext):
-    async with state.proxy() as data:
-        data['surname'] = message.text
+    @dp.message_handler(content_types=['contact'])
+    async def adding_to_db(message: types.Message):
         try:
-            database.register(f"@{message.from_user.username}", data['name'], data['surname'], message.from_user.id)
-            await message.answer('Vielen Dank fur die Registrierung!')
+            database.register(f"@{message.from_user.username}", message.from_user.first_name, message.from_user.id, message.contact['phone_number'])
```

Comment:
why do you still add to database by `message.from_user.username`, and what will happen then if the user has no username?

Reply:
nothing will happer cause this is not a key value in a database. what is more, it is not even an unique value, so a bot has multiple values to identify a person: by his username, name and contact. it's supposed that a user can't register without his/her contact: in a bot and in telegram as well